wormchain/x/wormhole/keeper/allowlist.go

```diff
@@ -0,0 +1,86 @@
+package keeper
+
+import (
+	"bytes"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/wormhole-foundation/wormchain/x/wormhole/types"
+)
+
+// SetSequenceCounter set a specific sequenceCounter in the store from its index
+func (k Keeper) SetValidatorAllowedAddress(ctx sdk.Context, address types.ValidatorAllowedAddress) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ValidatorAllowlistKey))
+	b := k.cdc.MustMarshal(&address)
+	store.Set([]byte(address.AllowedAddress), b)
+}
+
+func (k Keeper) GetValidatorAllowedAddress(ctx sdk.Context, address string) types.ValidatorAllowedAddress {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ValidatorAllowlistKey))
+	b := store.Get([]byte(address))
+	var allowedAddr types.ValidatorAllowedAddress
+	k.cdc.MustUnmarshal(b, &allowedAddr)
+	return allowedAddr
+}
+
+func (k Keeper) HasValidatorAllowedAddress(ctx sdk.Context, address string) bool {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ValidatorAllowlistKey))
+	return store.Has([]byte(address))
+}
+
+// RemoveSequenceCounter removes a sequenceCounter from the store
+func (k Keeper) RemoveValidatorAllowedAddress(ctx sdk.Context, address string) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ValidatorAllowlistKey))
+	store.Delete([]byte(address))
+}
+
+func (k Keeper) GetAllAllowedAddresses(ctx sdk.Context) (list []types.ValidatorAllowedAddress) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ValidatorAllowlistKey))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.ValidatorAllowedAddress
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+
+	return
+}
+
+// Checks if a given address is registered as a guardian validator and either:
+// * Is in the current guardian set, OR
+// * Is in a future guardian set
+func (k Keeper) IsAddressValidatorOrFutureValidator(ctx sdk.Context, addr string) bool {
```

Comment:
general question on this: what if a guardian is registered as a validator, and then the guardian is removed from the guardian set?
This function guards against that case. Should there additionally be some mechanism in wormchain that removes a validator if they are no longer part of the guardian set? (or does this mechanism already exist)?

Reply:
It does not currently exist, right now the root of trust is being in the guardian set or not.

Reply:
got it. So we're covered on all other paths here? Only other one I can think of is registering guardian as validator, but we're covered there